sotodlib/site_pipeline/update_smurf_caldbs.py

```diff
@@ -109,15 +123,252 @@ def smurf_detset_info(config:str | dict,
             })
         write_dataset(
             det_rs, 
-            config['archive']['h5file'], 
+            config['archive']['detset']['h5file'], 
             ts.name, 
             overwrite,
         )
         # add new entries to database
         if ts.name not in added_detsets:
             db_data = {'dets:detset': ts.name,
                     'dataset': ts.name}
-            db.add_entry(db_data, config['archive']['h5file'])
+            db.add_entry(db_data, config['archive']['detset']['h5file'])
+
+
+def get_cal_obsids(ctx, obs_id, cal_type):
+    """
+    Returns set of obs-ids corresponding to the most recent calibration
+    operations for a given obsid:
+
+    Returns
+    ----------
+        obs_ids: dict
+            Dict of obs_ids for each detset in specified operation
```

Comment:
Did you test the sphinx build for this? I think you want a period at the end of line 140 and one less indent on lines 144 & 145. Should also have a block for input args.